### DISCUSSION TITLE
defined nixci jobs in flake outputs not working

### DISCUSSION THREAD

**Discussion by tgunnoe** [srid/nixci#73] [Q&A]
Maybe I've read the docs incorrectly, but I thought I should be able to do this:
```nix
nixci = {
  builds = with inputs.self; {
    inherit packages devShells apps;
  };
  tests = with inputs.self; {inherit checks;};
};
```
`nixci build .#tests`

```
Error: Command error: Process exited unsuccessfully. exit_code=Some(1) stderr=warning: Git tree '/home//src/project' is dirty
error: flake 'git+file:///home/src/project' does not provide attribute 'packages.x86_64-linux.nixci.tests', 'legacyPackages.x86_64-linux.nixci.tests' or 'nixci.tests'


Caused by:
    Process exited unsuccessfully. exit_code=Some(1) stderr=warning: Git tree '/home/src/project' is dirty
    error: flake 'git+file:///home/src/project' does not provide attribute 'packages.x86_64-linux.nixci.tests', 'legacyPackages.x86_64-linux.nixci.tests' or 'nixci.tests'
```
I can't reference directly either:
```
nixci build .#nixci.x86_64-linux.tests
Error: Invalid flake URL (too many nested attr): .
```

**Comment by srid** [comment]
Not sure what you are trying to do. You don't need any `nixci` configuration for just building your top-level flake outputs. An explicit configuration would look like this:

```nix
{
  nixci.default = {
    "root".dir = ./.;
  };
}
```

(But that's the default anyway, when a nixci config is lacking).

**Reply by tgunnoe** [reply]
I am trying to exclude particular outputs, without having to explicitly define all of the ones I want. eg `nixci build .#packages.x86_64-linux.a .#devShells.x86_64-linux.b .#c .#e` ... 
First time using the config

**Reply by srid** [reply]
The config is for specifying which sub-flakes to build (see [example](https://github.com/juspay/services-flake/blob/ea3a18a991fd5e0df3543f17c9209a6942068c4a/flake.nix#L15-L41)), and not which flake outputs to build.

`nixci` will build _all_ flake outputs. There is (currently) no way to blacklist that. You can however whitelist systems, like [here](https://github.com/srid/nixos-flake/blob/face52ad2209fa6c50c7c0b88e1e71d177e0485b/flake.nix#L29-L49).

**Reply by tgunnoe** [reply]
ok thanks buddy. Maybe i'll look at a PR for blacklisting outputs if you're interested.
